### PR TITLE
Replaced object-hash with hash-sum to hopefully get a more determinis…

### PR DESCRIPTION
**Pull request by JakeGinnivan**
…tic hash

### CODE REVIEW

File: tests/__snapshots__/data-loader.csr.spec.tsx.snap [hunk 1]
```diff
@@ -35,7 +35,7 @@ Object {
     },
     Object {
       "state": Object {
-        "fc7e58063ef49874aea8e2210fc7a6a2f9836f52": Object {
```
Comment (JakeGinnivan) [inline tests/__snapshots__/data-loader.csr.spec.tsx.snap]: This is basically the issue, this hash `should` be the same on client and server, and it is for all browsers except IE11. This new library is far simpler, smaller and hopefully faster.

File: src/data-provider.tsx [hunk 2]
```diff
@@ -2,14 +2,20 @@ import React from 'react'
...
+    ObjectHash,
+} from './data-loader-store-and-loader'
 
 export interface Props {
     initialState?: DataLoaderState
     onEvent?: (event: DataProviderEvents) => void | Promise<any>
     isServerSideRender?: boolean
     resources: DataLoaderResources<any>
     globalProps?: object
+    /** Override the object hasing function */
```
Comment (unique3) [inline src/data-provider.tsx]: hashing

File: tests/__snapshots__/data-loader.csr.spec.tsx.snap (hunk 1 above)
Comment (mandymichael) [reply tests/__snapshots__/data-loader.csr.spec.tsx.snap]: Will be interested to see if this fixes some of the random issues we've had in IE11 between the SSR and client side render.

**Pull request merged**